src/hotspot/share/opto/type.cpp

```diff
@@ -2403,6 +2410,15 @@ const TypeVect* TypeVect::make(const Type *elem, uint length) {
   return NULL;
 }
 
+const TypeVect *TypeVect::makemask(const Type* elem, uint length) {
+  if (Matcher::has_predicated_vectors()) {
+    const TypeVect* mtype = Matcher::predicate_reg_type(elem, length);
+    return (TypeVect*)(const_cast<TypeVect*>(mtype))->hashcons();
+  } else {
+    return make(elem, length);
+  }
```

Comment:
For non predicated arch, do we expect this function to be called? Or at least should we assert elem->array_element_basic_type() is a boolean for the else{} block?

Reply:
Yes, for non-predicated architectures it will allocate a regular vector type.